lib/foreman_deployments/dynflow_output_patches.rb

```diff
@@ -0,0 +1,19 @@
+module Dynflow
+  module ForemanDeploymentsExtensions
+    module OutputReferenceExtensions
+      def dereference(persistence)
+        action_data = persistence.adapter.load_action(execution_plan_id, action_id)
+        action = action_data[:class].constantize
+        if action.respond_to? :dereference_output
+          action.dereference_output(action_data, @subkeys)
+        else
+          super
+        end
+      end
+    end
+  end
+
+  class ExecutionPlan::OutputReference
+    prepend Dynflow::ForemanDeploymentsExtensions::OutputReferenceExtensions
+  end
+end
```

Comment:
@iNecas could you please check the code from dynflow perspective? I know there was some work planned around active record de-serialization. Has there been any progress in that area? Is this a correct temporary solution or are there better ways? Thanks